jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -204,6 +204,10 @@ def _return_spectrum_with_correct_units(flux, wcs, metadata, data_type=None,
     if (apply_pix2 and (data_type != "mask") and
             (not check_if_unit_is_per_solid_angle(flux.unit))):
         target_flux_unit = flux.unit / PIX2
+    elif check_if_unit_is_per_solid_angle(flux.unit, return_unit=True) == "spaxel":
+        # We need to convert spaxel to pixel squared, since spaxel isn't fully supported by astropy
+        # This is horribly ugly but just multiplying by u.Unit("spaxel") doesn't work
+        target_flux_unit = flux.unit * u.Unit('spaxel') / PIX2
```

Comment:
@cshanahan1 just pointing out that this is where `spaxel` is getting converted to `PIX2` in the parser - as discussed offline we shouldn't need spaxel handling elsewhere.